core/src/main/java/tc/oc/pgm/match/MatchPlayerImpl.java

```diff
@@ -433,6 +434,18 @@ public void tick(Match match, Tick tick) {
     return audience;
   }
 
+  @Nullable
+  @Override
+  public MatchPlayer getSpectatorTarget() {
+    if (getBukkit() == null) return null;
+    return match.getPlayer(getBukkit().getSpectatorTarget());
```

Comment:
Looks like a race condition.

```suggestion
    Player bukkit = getBukkit();
    return bukkit == null ? null : match.getPlayer(bukkit.getSpectatorTarget());
```